ast/builtins.go

```diff
@@ -720,6 +722,28 @@ var URLQueryEncode = &Builtin{
 	),
 }
 
+// URLQueryFromJSON encodes the given JSON into a URL encoded query string
+var URLQueryFromJSON = &Builtin{
+	Name: "urlquery.fromjson",
+	Decl: types.NewFunction(
+		types.Args(types.A),
+		types.S,
+	),
+}
+
+// URLGenerate takes a host and a path and a list of query arguments in JSON format and encodes it into a string
+var URLGenerate = &Builtin{
```

Comment:
I'm hesitant to add this built-in function for two reasons.

1. It assumes that users don't need to control the scheme or the port. We could extend it to support those, but what about other cases? E.g., sometimes you may not want the URL parameters encoded.

2. It's possible to implement both fo these new built-in functions in Rego.